ssl/t1_lib.c

```diff
@@ -2463,8 +2559,10 @@ int tls_choose_sigalg(SSL *s, int fatalerrs)
     }
     if (sig_idx == -1)
         sig_idx = lu->sig_idx;
-    s->s3->tmp.cert = &s->cert->pkeys[sig_idx];
-    s->cert->key = s->s3->tmp.cert;
+    if (s->s3->tmp.cert == NULL) {
```

Comment:
Is this change still correct? I'm not sure why its needed.

Reply:
It is not needed, though it should also not be incorrect (since ```s->s3->tmp.cert``` should always be NULL).